docs/metrics.asciidoc

```diff
@@ -60,3 +60,69 @@ This value is normalized by the number of CPU cores and it ranges from 0 to 100%
 
 The Resident Set Size,
 the amount of memory the process occupies in main memory (RAM).
+
+[float]
+[[metric-nodejs.active_handles]]
+=== `nodejs.active_handles`
```

Comment:
> Are these metrics that people typically use for diagnosing Node.js performance issues? What would you do with this information? They seem pretty low level, leaking internal details of libuv.

Each number viewed without context doesn't really tell you anything, but it's the trend that's interesting. If `active_handles` starts to increase over time, it means you're putting more and more load on the system. And if `active_requests` rise over time, it could be an indicator of a leak of some sorts where requests are not being terminated for instance.

`active_handles` are 99% file descriptors. So a rise in this number means that you're consuming more and more file descriptors.

But yes, they are low level. And while you probably could get more high-level numbers, they are super easy and zero cost for us to collect. So it's definitely worth it.

> Since they are related to libuv, should active_handles and active_requests come under nodejs.eventloop?

The event loop is not the same as libuv. Part of it is inside of libuv (others not), but libuv does a lot of other things as well (most notably I/O).

Reply:
OK. My question rises from not being familiar with the terms "handles" and "requests" in the context. If it's common knowledge what an "active handle" or "active request" is, no problem. Otherwise maybe it should be nested under `nodejs.libuv`. I'll butt out now :)